### DISCUSSION TITLE
Scaling Gateway Resources

### DISCUSSION THREAD

**Discussion by shaneutt** [kubernetes-sigs/gateway-api#1355] [General]
At a community sync I posited the idea of having specification that would enable scaling `Gateway` resources. In the discussion it appeared at least two implementations already have some kind of functionality or provision for this currently using custom means. That is to say: some mechanism for scaling up or down the number of instances of the underlying gateway (or proxy, if you will). The purpose of this discussion is to ask the community to share if and how their implementations handle this currently and also just generally discuss the downsides and upsides of trying to put something like this into the spec. The purpose of collecting this feedback is to inform a possible future in-tree solution _or_ as posterity for the reasons why we do not have an in-tree solution.

**Comment by bowei** [comment]
As mentioned on the call, using K8s primitives to do this kind of autoscaling for a self-managed Gateway in-cluster implementation would be an interesting Kubecon talk!

Couple of things:

* This seems to be a generic problem that is not specific to GW -- it applies to any traffic-serving infra. 
* Ideal end state for K8s infra is to be auto-scaled and "smart" -- do we really want to formalize knobs for the user if that is the end goal?

**Comment by tomkerkhove** [comment]
Have you considered delegating the autoscaling of Gateway resources to KEDA? It allows you to [autoscale custom resources](https://keda.sh/docs/latest/concepts/scaling-deployments/#scaling-of-custom-resources) and you just have to implement `/scale` subresource.

**Reply by shaneutt** [reply]
I had not considered this yet, I'll dig in thank you for sharing :partying_face:

**Reply by tomkerkhove** [reply]
Happy to help! I'm one of the maintainers so feel free to ping me if you want to chat.
